src/messages/mod.rs

```diff
@@ -170,6 +169,48 @@ impl Message {
     pub fn hash(&self) -> &MessageHash {
         &self.hash
     }
+
+    // Extend the current message proof so it starts at `new_first_key` while keeping the last key
+    // (and therefore the signature) intact.
+    #[cfg_attr(feature = "mock_base", allow(clippy::trivially_copy_pass_by_ref))]
```

Comment:
Just a quick one, do we need this flag here?

Reply:
Because `feature=mock_base` also enables the mock version of threshold-crypto and with it the `bls::PublicKey` type is only 4 bytes long. And clippy things that for types that small, it's more efficient to pass them by value. But with the normal production version of threshold-crypto, the type has ~40 bytes so it's definitely more efficient to pass it by reference.

Reply:
Be nice from yesterdays meet to see just how much of a hitr it is here to not use mock threshold crypto. Esepecially so as the dkg PR is en route.

Reply:
As far as I remember, it was pretty significant. But we can try to measure it again.

Reply:
Yes thanks @madadam  For sure crypto is slow and bls is a factor slower than ed crypto. So likely they will be treacle in dev mode but should be acceptable in release mode.